pkg/analytics/instrumentation_collector.go

```diff
@@ -123,23 +125,53 @@ func (ic *instrumentationCollectorImpl) AddExtension(key string, value interface
 
 func GetV2InstrumentationObject(collector InstrumentationCollector) (*api.AnalyticsRequestBody, error) {
 	t, ok := collector.(*instrumentationCollectorImpl)
-	if ok {
-		return t.GetV2InstrumentationObject(), nil
+	if !ok {
+		return nil, fmt.Errorf("failed to convert collector")
 	}
-	return nil, fmt.Errorf("failed to convert collector")
+
+	return t.GetV2InstrumentationObject()
 }
 
-func (ic *instrumentationCollectorImpl) GetV2InstrumentationObject() *api.AnalyticsRequestBody {
+func (ic *instrumentationCollectorImpl) GetV2InstrumentationObject() (*api.AnalyticsRequestBody, error) {
 	a := ic.getV2Attributes()
 
 	d := api.AnalyticsData{
 		Type:       ic.instrumentationType,
 		Attributes: a,
 	}
 
+	// Since the `extension` attribute in the analytics payload is a value any
```

Comment:
Question: I'm just thinking a bit about the error handling here. if the sanitization fails for whatever reason, we might want to still send the other data and just skip the extension data?

Reply:
Yes, good point. Will fix.

Reply:
Update on this for reviewers: We synced offline and decided we don't have a strong opinion on an approach between:

Fail hard
Fail semi-hard with type assertions on different errors higher in the call stack, but having to default to string comparisons
Fail soft but pass a logger object (thus causing a change in the SDK API requiring downstream changes to other libs)
Current decision: Make a new API endpoint WithLogger or something of that sort to avoid breaking changes, possibly update snyk-ls to use this if IDE agrees.
Fail soft, log nothing, lose the context in debug logs